src/dotnet-interactive/Http/HttpProbingSettings.cs

```diff
@@ -9,53 +9,49 @@
 
 namespace Microsoft.DotNet.Interactive.Http;
 
-internal class HttpProbingSettings
+public class HttpProbingSettings
 {
-    public Uri[] AddressList { get; private set; }
+    public IEnumerable<string> AddressList { get; private set; }
 
-    public static HttpProbingSettings Create(int? port)
+    public static HttpProbingSettings Create(int? httpPort, bool httpLocalOnly)
     {
+        HashSet<string> ipAddress = [IPAddress.Loopback.ToString()];
+
+        if (!httpLocalOnly)
+        {
+            ipAddress = GetAllNetworkInterfaces()
+                .Where(ni => ni.OperationalStatus == OperationalStatus.Up)
+                .SelectMany(ni => ni.GetIPProperties().UnicastAddresses)
+                .Select(x => x.Address.ToString())
+                .Concat(ipAddress)
+                .ToHashSet();
+        }
+
+        var uriAddresses = ipAddress
+            .Select(AddHttpPort(httpPort));
+
         return new HttpProbingSettings
         {
-            AddressList = GetProbingAddressList(port)
+            AddressList = uriAddresses
         };
     }
 
-    private static Uri[] GetProbingAddressList(int? httpPort)
+    private static Func<string, string> AddHttpPort(int? httpPort)
     {
-        var sources = new List<string>();
-        foreach (var ni in NetworkInterface.GetAllNetworkInterfaces())
-        {
-            if (ni.OperationalStatus == OperationalStatus.Up)
-            {
-                foreach (var ip in ni.GetIPProperties().UnicastAddresses.Select(a => a.Address.ToString()))
-                {
-
-                    if (ip != IPAddress.Loopback.ToString())
-                    {
-                        sources.Add(ip);
-                    }
-                }
-            }
-        }
+        if (httpPort is null)
+            return ipAddress => $"http://{ipAddress}/";
 
-        sources.Add(IPAddress.Loopback.ToString());
+        return ipAddress => $"http://{ipAddress}:{httpPort}/";
+    }
 
-        var addresses = sources
-            .Where(s => !string.IsNullOrWhiteSpace(s))
-            .Select(s =>
-            {
-                var uriString = httpPort is not null ? $"http://{s}:{httpPort}/" : $"http://{s}/";
-                if (Uri.TryCreate(uriString, UriKind.Absolute, out var uri))
-                {
-                    return uri;
-                }
+    // Delegate that matches the signature of GetAllNetworkInterfaces
+    public delegate NetworkInterface[] GetAllNetworkInterfacesDelegate();
```

Comment:
The custom delegate isn't needed here since it's such a simple type, e.g. it has a single use and isn't a complex enough delegate type to provide some self-documentation benefits from named parameters, etc. 

So it's preferable to use `Func<NetworkInterface[]>`. We're generally trying to reduce the number of types on the way to GA.